File: tournaments/hangman/hangman.py
def hangman(word, letters):
    result = 0
    a = 0

    for item in letters:
        if 6 <= a:
            return False
        b = word.count(item)
        if 0 == b:
            a += 1
        else:
            result += b

    return len(word) == result

File: tournaments/hangman/hangman.js
function hangman(word, letters) {
    const store = new Array(26).fill(0);
    const aCode = 'a'.charCodeAt();
    let result = 0;
    let a = 0;

    for (const item of word) {
        ++store[item.charCodeAt() - aCode];
    }

    for (const item of letters) {
        if (6 <= a) {
            return false;
        }

        const b = store[item.charCodeAt() - aCode];

        if (!b) {
            ++a;
        } else {
            result += b;
        }
    }

    return word.length === result;
}
